### PR TITLE
WIP utils: add get_record_pid_type helper

### LINKED ISSUE #1355: schema utilities
- [ ] We need a function that given a JSON parses from the schema URL the document type
- [ ] We need a function that given document type returns the Schema URL to be put in the schema field.
- [ ] We should refactor to use above functions.

### PR DESCRIPTION
**Pull request by jacquerie**
Will also close #1355 and these:

```
% ag \#1355
inspirehep/modules/disambiguation/receivers.py
98:    # FIXME: Use a dedicated method when #1355 will be resolved.
113:    # FIXME: Use a dedicated method when #1355 will be resolved.

inspirehep/modules/disambiguation/records.py
49:    # FIXME: Use a dedicated method when #1355 will be resolved.
```

Closes #1570

### CODE REVIEW

File: tests/unit/utils/test_utils_record.py [hunk 1]
```diff
@@ -294,3 +300,30 @@ def test_get_value_returns_none_on_index_error():
...
+
+@pytest.mark.xfail(reason='does not know which schemas really exist.')
+def test_get_record_pid_type_raises_error_for_nonexisting_schema():
+    record = Record({
+        '$schema': 'does-not-exist.json'
+    })
+
+    with pytest.raises(ValueError) as excinfo:
+        get_record_pid_type(record)
+    assert str(excinfo.value) == 'This record has an invalid schema.'
```
Comment (jacquerie) [inline tests/unit/utils/test_utils_record.py]: This proposed API should address https://github.com/inspirehep/inspire-next/pull/1570#discussion_r79777420.

Comment (jmartinm) [inline tests/unit/utils/test_utils_record.py]: 👍  So you are going to add that error message to `get_record_pid_type` on this PR?

Comment (jacquerie) [inline tests/unit/utils/test_utils_record.py]: That was the idea, but I'm waiting for #1613 (which is waiting for some changes upstream in `invenio-records-rest`) to see if this PR will continue to work.

**Comment by jacquerie** [comment]
Superseded by #1680.

**Pull request closed without merging**